### PR TITLE
add initial corpus

### PR DESCRIPTION
**Pull request by rogpeppe**
We use the new line-protocol/v2 package to generate
the test case outputs, as that's the implementation
most trusted currently.

### CODE REVIEW

File: schema.cue [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+#Precision: "1ns" | "1µs" | "1ms" | "1s" | "1m" | "1h"
+
+#Point: {
+	time: int
+	name: #StringOrBinary
+	tags: [... #Tag]
+	fields: [... #Field]
+}
+
+foo: #Point & {
```
Comment (alamb) [inline schema.cue]: I probably don't understand this, but I found the `foo` here confusing

**Review by alamb — COMMENTED** [review]
I read the documentation, downloaded the code, and poked around. Looks good to me 👍 

Comments:
1. I have not run across cue before -- it looks fairly `golang` centric. It isn't clear to me how I would use this corpus to test the rust parser (I would have to sort out how to parse cue in Rust and then hook up a test harness?) so perhaps some comments in this regard would help

2. It would help me if there was some documentation about what the individual corpuses are / represent.. For example what is the difference between `inputs-decode-custom.cue` and ` inputs-decode-fuzz.cue`. Maybe this is explained in the linked doc and could just be copy / pasted?

3. I wonder if you add a brief example of "How to run this code" or "how to use this code" -- it is probably obvious to a Go expert, but it took me some time and poke around. For example, I needed to install `cue` via `GO111MODULE=on go get cuelang.org/go/cmd/cue` and then `go run generate.go`


Once I did run the code it looks like what is created for me is not the same as what is checked into this PR. This might be ok, but I wanted to point it out

```
(parquet) alamb@ip-10-0-0-124:~/Software/line-protocol-corpus$ git status
On branch 001-initial-commit
Your branch is up to date with 'origin/001-initial-commit'.

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   corpus-decode-generated.cue
	modified:   corpus-encode-generated.cue
	deleted:    corpus.json
	modified:   go.mod
	modified:   go.sum

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	line-protocol-corpus

no changes added to commit (use "git add" and/or "git commit -a")
(parquet) alamb@ip-10-0-0-124:~/Software/line-protocol-corpus$ git diff
diff --git a/corpus-decode-generated.cue b/corpus-decode-generated.cue
index 1a5ee94..8a66269 100644
--- a/corpus-decode-generated.cue
+++ b/corpus-decode-generated.cue
@@ -44,7 +44,7 @@ corpus: decode: {
        }
        "0025cd7e54e673041e5abd93443c2dea": {
                input: {
-                       text:           "m\v2=\",\"\v58259079175675784\t="
+                       text:           "m\u000b2=\",\"\u000b58259079175675784\t="
                        defaultTime:    946815194000000000
                        precision:      "1ns"
                        implementation: "fuzz"
@@ -53,7 +53,7 @@ corpus: decode: {
        }
        "00393286f265d103f438a7dee343ea93": {
                input: {
-                       text:           "m\v2=\"\v2=FALS"
+                       text:           "m\u000b2=\"\u000b2=FALS"
                        defaultTime:    946815194000000000
                        precision:      "1ns"
                        implementation: "fuzz"
@@ -62,7 +62,7 @@ corpus: decode: {
```

**Review by alamb — APPROVED** [review]

File: schema.cue (hunk 1 above)
Comment (rogpeppe) [reply schema.cue]: It's a debugging remnant. You were right to be confused.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+reproduce that exact error.
+
+The corpus results are intended correspond with the proposed standardised syntax
+[proposal](https://docs.google.com/document/d/1GCGgg6EcW8bPyVhLsS9L6kvEyC8S5_T0tbcomfF3FxI).
+
+## Using the corpus
+
+To use this corpus to run against your own codec, you don't need to install CUE,
+as all the relevant data is exported into the [corpus.json](corpus.json) file.
+You'll need to write some code to translate from the unmarshaled JSON to the data
```
Comment (alamb) [inline README.md:26]: 👍

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+CUE_FILES=$(wildcard *.cue)
+CORPUS_FILES=corpus-decode-generated.cue corpus-encode-generated.cue
+
+corpus.json: $(CORPUS_FILES)
+	rm -f corpus.json
+	cue export --out json -e corpus -o corpus.json
+
+$(CORPUS_FILES): $(filter-out $(CORPUS_FILES),$(CUE_FILES))
+	rm -f $(CORPUS_FILES)
+	go run generate.go
```
Comment (alamb) [inline Makefile:13]: ❤️

**Review by alamb — APPROVED** [review]
Looks like a nice improvement --  thanks @rogpeppe

**Pull request merged**